internal/controller/confluent/apikey/zz_controller.go

```diff
@@ -30,9 +35,11 @@ func Setup(mgr ctrl.Manager, o tjcontroller.Options) error {
 	if o.SecretStoreConfigGVK != nil {
 		cps = append(cps, connection.NewDetailsManager(mgr.GetClient(), *o.SecretStoreConfigGVK, connection.WithTLSConfig(o.ESSOptions.TLSConfig)))
 	}
+	eventHandler := handler.NewEventHandler(handler.WithLogger(o.Logger.WithValues("gvk", v1alpha1.APIKey_GroupVersionKind)))
+	ac := tjcontroller.NewAPICallbacks(mgr, xpresource.ManagedKind(v1alpha1.APIKey_GroupVersionKind), tjcontroller.WithEventHandler(eventHandler))
 	opts := []managed.ReconcilerOption{
-		managed.WithExternalConnecter(tjcontroller.NewConnector(mgr.GetClient(), o.WorkspaceStore, o.SetupFn, o.Provider.Resources["confluent_api_key"], tjcontroller.WithLogger(o.Logger),
-			tjcontroller.WithCallbackProvider(tjcontroller.NewAPICallbacks(mgr, xpresource.ManagedKind(v1alpha1.APIKey_GroupVersionKind))),
+		managed.WithExternalConnecter(tjcontroller.NewConnector(mgr.GetClient(), o.WorkspaceStore, o.SetupFn, o.Provider.Resources["confluent_api_key"], tjcontroller.WithLogger(o.Logger), tjcontroller.WithConnectorEventHandler(eventHandler),
```

Comment:
Since all of these `zz_controller.go` files look very similar, can we instead have one file that serves as a template for all of these files? Then, these files can simply call a function with a parameter representing what folder they are currently in (e.g. `apikey`, `cluster`, `clusterconfig`, etc.). Based on that parameter, there can be a struct/map that defines all the variables that need to be changed and what their value is. So if the parameter is `apikey`, then the struct would map all of its values to anything used in that file (under `apikey/`) that is not in any other `zz_controller.go` file. For example, this file specifically uses `"confluent_api_key"` on line 41, so one of the fields in the struct/map will have its value as `"confluent_api_key"`, but in another instance of the struct it would be a different value corresponding to whatever string was used in place of `"confluent_api_key"` on line 41.

Reply:
@Chrand123 It's not a bad idea for reducing code duplicity, but these files are generated code that we cannot control.

See https://github.com/crossplane/upjet for more info